frontend/hooks/useNeedsFunds.ts

```diff
@@ -1,54 +1,114 @@
+  const { storeState } = useStore();
+  const { service } = useService({ serviceConfigId });
+  const { masterSafes } = useMasterWalletContext();
+  const { isLoaded: isBalanceLoaded, walletBalances } =
+    useBalanceContext();
+    
+
+  const isInitialFunded = storeState?.isInitialFunded;
 
-  const serviceTemplate = useMemo(
-    () => getServiceTemplates()[0],
-    [getServiceTemplates],
+  const serviceTemplate = useMemo<ServiceTemplate | undefined>(
+    () => (service ? getServiceTemplate(service.hash) : undefined),
+    [service],
   );
 
-  const { storeState } = useStore();
-  const isInitialFunded = storeState?.isInitialFunded;
+  const serviceFundRequirements = useMemo< {
+    [chainId: number]: {
+      [tokenSymbol: string]: number;    
+    }
+  }>(() => {
+    if (!serviceTemplate) return {};
 
-  const {
-    isBalanceLoaded,
-    masterSafeBalance: safeBalance,
-    totalOlasStakedBalance,
-  } = useBalanceContext();
-
-  const serviceFundRequirements = useMemo(() => {
-    const gasEstimate =
-      serviceTemplate.configurations[CHAIN_CONFIG.OPTIMISM.chainId]
-        .monthly_gas_estimate;
-    const monthlyGasEstimate = Number(formatUnits(`${gasEstimate}`, 18));
-    const minimumStakedAmountRequired =
-      getMinimumStakedAmountRequired(serviceTemplate);
-
-    return { eth: monthlyGasEstimate, olas: minimumStakedAmountRequired };
+    const results: {
+      [chainId: number]: {
+        [tokenSymbol: string]: number;    
+      }
+    } = {};
+
+    Object.entries(serviceTemplate.configurations).forEach(
+      ([chainId, config]) => {
+        const serviceTemplateDefault = serviceTemplate.configurations[+chainId].staking_program_id
+        const serviceCurrent = service?.chain_configs[+chainId]?.chain_data?.user_params?.staking_program_id
+
+        if (!serviceCurrent && !serviceTemplateDefault) return;
+
+        if (!service?.chain_configs[+chainId]) return;
+        const gasEstimate = config.monthly_gas_estimate;
+        const monthlyGasEstimate = Number(formatUnits(`${gasEstimate}`, 18));
+        const minimumStakedAmountRequired =
+          STAKING_PROGRAMS[+chainId][
+            service?.chain_configs[+chainId]?.chain_data?.user_params
+              ?.staking_program_id ??
+              serviceTemplate.configurations[+chainId].staking_program_id
+          ].stakingRequirements.OLAS;
+
+        const nativeTokenSymbol = getNativeTokenSymbol(+chainId);
+
+        results[+chainId] = {
+          [TokenSymbol.OLAS]: +formatEther(minimumStakedAmountRequired),
+          [nativeTokenSymbol]: +formatEther(monthlyGasEstimate),
+          // TODO: extend with any further erc20s..
+        };
+      },
+    );
+
+    return results;
   }, [serviceTemplate]);
 
   const hasEnoughEthForInitialFunding = useMemo(
-    () => (safeBalance?.ETH || 0) >= (serviceFundRequirements?.eth || 0),
-    [serviceFundRequirements?.eth, safeBalance],
+    () => {
+      if (!serviceFundRequirements) return ;
```

Comment:
also small, but I noticed in `needsInitialFunding` you return false in similar cases, but here undefined;
for type safety and consistency, maybe you could specify the return type? might help prevent surprises
```
const hasEnoughEthForInitialFunding: boolean = useMemo(
```

Reply:
yeah, i agree, however, boolean hides that the value is not yet loaded..
i.e. if it's true/false we can't assume that a new value is coming, or it is not yet defined